### PR TITLE
Added Feature to inspect node by Chef::Platform.windows? 

### PR DESCRIPTION
**Pull request by piyushawasthi**
- Added Feature to inspect node by Chef::Platform.windows?
- Re-factor all codes and specs related to Chef::Platform.windows?
- Re-factor code of provider_mapping.rb.
- Fixed failing specs. 
wrt - https://github.com/chef/chef/issues/5771

### CODE REVIEW

File: chef-config/lib/chef-config/windows.rb [hunk 1]
```diff
@@ -17,8 +17,9 @@
 
 module ChefConfig
   def self.windows?
-    if @node
-      platform = @node[:platform] || @node[:platform_family] || @node[:os]
+    @node ||= nil
+    if @node && @node[:os]
```
Comment (smurawski) [inline chef-config/lib/chef-config/windows.rb]: Wouldn't this always result in returning `true` if @node is present and has an :os key, even on non-windows systems?

File: chef-config/lib/chef-config/windows.rb [hunk 2]
```diff
@@ -16,13 +16,18 @@
 #
 
 module ChefConfig
-
   def self.windows?
-    if RUBY_PLATFORM =~ /mswin|mingw|windows/
+    @node ||= nil
+    if @node && @node[:os]
```
Comment (smurawski) [inline chef-config/lib/chef-config/windows.rb]: Wouldn't this always result in returning true if @node is present and has an :os key, even on non-windows systems?

Comment (piyushawasthi) [inline chef-config/lib/chef-config/windows.rb]: The node[:os] only present for windows

**Comment by NimishaS** [comment]
@btm , @lamont-granquist , @smurawski , @mwrock , please check the comments given by @coderanger in https://github.com/chef/chef/pull/6048.

**Comment by btm** [comment]
We're going to wait for Lamont to come back from vacation and work this into his plans for [rfc087](https://github.com/chef/chef-rfc/blob/master/rfc087-distro-sugar-helpers.md). I'm to close this PR and we'll keep the issue #5771 open to track further progress.

**Pull request closed without merging**